### DISCUSSION TITLE
Context manager time out while creating tabs

### DISCUSSION THREAD

**Discussion by grnsingh** [zauberzeug/nicegui#3430] [Q&A]
### Question

I am trying to build a browser styled data analysis tool, with a home page for data collection, and then add tab button creates a new tab with relevant data, the issue is that  this relevant data takes 5-6 seconds to parse, and store. (Bought it down from 28 to 5 using processes can't go further down) 

But still due to 5-6 seconds load time nicegui misses to add this tab and panel, is there a way to create the child element first and then add to to the parent (in this case the tab) or is there a way to increase the context managers timeout ?

**Comment by python-and-novella** [comment]
you should do it in subprocess instead of main process.

**Comment by rodja** [comment]
You can [move elements around](https://nicegui.io/documentation/element#move_elements). But @python-and-fiction is right. Please make sure you do not any cpu intensive tasks on the main thread: https://github.com/zauberzeug/nicegui/wiki/FAQs#why-is-my-long-running-function-blocking-ui-updates
